das_api/src/api/api_impl.rs

```diff
@@ -397,9 +397,18 @@ impl ApiContract for DasApi {
             page,
             before,
             after,
+            tree,
+            leaf_index,
         } = payload;
-        let id = validate_pubkey(id.clone())?;
-        let id = id.to_bytes().to_vec();
+
+        if id.is_none() && (tree.is_none() || leaf_index.is_none()) {
```

Comment:
I think this should also fail if all 3 are provided, right?

Reply:
My understanding is it shouldn't, since we have all the data we need to return a valid response.

Reply:
What if the asset id does not match the expected leaf? Imo it should be one or the other